wallet/src/main/scala/org/bitcoins/wallet/internal/TransactionProcessing.scala

```diff
@@ -419,30 +457,38 @@ private[bitcoins] trait TransactionProcessing extends WalletLogger {
 
     val processTxF = for {
       receivedSpendingInfoDbs <- receivedSpendingInfoDbsF
-      receivedStart = TimeUtil.currentEpochMs
-      incoming <- processReceivedUtxos(
+      incomingA = processReceivedUtxos(
         transaction = transaction,
         blockHashOpt = blockHashOpt,
+        confirmations = confirmations,
         spendingInfoDbs = receivedSpendingInfoDbs,
         newTags = newTags,
         relevantReceivedOutputs = relevantReceivedOutputs
       )
-      _ = if (incoming.nonEmpty) {
-        logger.info(
-          s"Finished processing ${incoming.length} received outputs, balance=${incoming
-            .map(_.output.value)
-            .sum} it took=${TimeUtil.currentEpochMs - receivedStart}ms")
-      }
 
       spentSpendingInfoDbs <- spentSpendingInfoDbsF
-      spentStart = TimeUtil.currentEpochMs
-      outgoing <- processSpentUtxos(transaction = transaction,
+      outgoingA = processSpentUtxos(transaction = transaction,
                                     outputsBeingSpent = spentSpendingInfoDbs,
                                     blockHashOpt = blockHashOpt)
-      _ = if (outgoing.nonEmpty) {
+      tuple = DBIO.sequence(Vector(incomingA, outgoingA))
+      insertResult <- safeDatabase.run(tuple)
+      incoming = insertResult.head
+      outgoing = insertResult(1)
```

Comment:
this is kinda hacky, would be cleaner to do

```
incomingA.flatMap(incoming => outgoingA.map(incoming, _))
```

Since it is a database transaction it shouldn't change anything in performance